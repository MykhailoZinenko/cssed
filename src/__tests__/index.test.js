import pluginTester from 'babel-plugin-tester'
import plugin from '../../lib/index'

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
    parserOpts: {
      presets: [
        [
          'env',
          {
            modules: 'commonjs'
          }
        ],
        'react'
      ],
      plugins: ['jsx']
    }
  },
  tests: [
    {
      title: 'single call macro',
      code: `
        import { css } from 'cssed';

        const styles = css\`
          .box {
            color: blue
          }
        \`
      `
    },
    {
      title: 'single call evaluation',
      code: `
        import { css } from 'cssed';

        const blue = 'blue'

        const styles = css\`
          .box {
            color: \${blue}
          }
        \`
      `
    },
    {
      title: 'multi call macro',
      code: `
        import { css } from 'cssed';

        const blue = css\`
        .box {
          color: blue;
        }
        \`;

        const red = css\`
        .box {
          color: red;
        }
        \`;

        export default props => (
          <div className={props.isRed ? red.box : blue.box} />
        );
      `
    },
    {
      title: 'multi call with external dependency',
      code: `
        import { css } from 'cssed';
        import { light, dark } from './module'

        const btn = css\`
        .light {
          color: \${light};
        }
        .dark {
          color: \${dark};
        }
        \`;

        export default props => (
          <>
            <div className={btn.light} />
            <div className={btn.dark} />
          </>
        );
      `
    },
    {
      title: 'multi call with external module dependency',
      code: `
        import { css } from 'cssed';
        import { light, dark } from './module'

        const btn = css\`
        .light {
          color: \${light};
        }
        .dark {
          color: \${dark};
        }
        \`;

        export default props => (
          <>
            <div className={btn.light} />
            <div className={btn.dark} />
          </>
        );
      `
    },
    {
      title: 'function call in expression',
      code: `
        import { css } from 'cssed';
        import { light, dark } from './constant'

        const darkOrLight = (bool) => bool ? light : dark
        const btn = css\`
        .light {
          color: \${darkOrLight(true)};
        }
        .dark {
          color: \${darkOrLight(false)};
        }
        \`;

        export default props => (
          <>
            <div className={btn.light} />
            <div className={btn.dark} />
          </>
        );
      `
    },
    {
      title: 'test url rebase',
      code: `
        import { css } from 'cssed';

        const btn = css\`
        .light {
          background-image: url("star.gif");
          list-style-image: url('../images/bullet.jpg');
          font-family: 'Open Sans', sans-serif;
          border-image: url('border.png');
          content: url('/content.png');
          content: url(data:image/png;base64,iVBORw0KGgoAAAA);
        }
        \`;

        export default props => (
          <>
            <div className={btn.light} />
            <div className={btn.dark} />
          </>
        );
      `
    }
  ]
})
