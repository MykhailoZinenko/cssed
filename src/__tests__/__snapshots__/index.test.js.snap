// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros function call in expression: function call in expression 1`] = `

import { css } from '../../lib/macro';
import { light, dark } from './constant'

const darkOrLight = (bool) => bool ? light : dark
const btn = css\`
.light {
  color: \${darkOrLight(true)};
}
.dark {
  color: \${darkOrLight(false)};
}
\`;

export default props => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_a from './.index.test.module.css'
import { light, dark } from './constant'

const darkOrLight = (bool) => (bool ? light : dark)

const btn = _cssed_a
export default (props) => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
)


`;

exports[`macros multi call macro: multi call macro 1`] = `

import { css } from '../../lib/macro';

const blue = css\`
.box {
  color: blue;
}
\`;

const red = css\`
.box {
  color: red;
}
\`;

export default props => (
  <div className={props.isRed ? red.box : blue.box} />
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_b from './.index.test.b.module.css'
import _cssed_a from './.index.test.a.module.css'
const blue = _cssed_a
const red = _cssed_b
export default (props) => <div className={props.isRed ? red.box : blue.box} />


`;

exports[`macros multi call with external dependency: multi call with external dependency 1`] = `

import { css } from '../../lib/macro';
import { light, dark } from './constant'

const btn = css\`
.light {
  color: \${light};
}
.dark {
  color: \${dark};
}
\`;

export default props => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_a from './.index.test.module.css'
import { light, dark } from './constant'
const btn = _cssed_a
export default (props) => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
)


`;

exports[`macros single call macro: single call macro 1`] = `

import { css } from '../../lib/macro';

const styles = css\`
  .box {
    color: blue
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_a from './.index.test.module.css'
const styles = _cssed_a


`;

exports[`unknown plugin function call in expression: function call in expression 1`] = `

import { css } from '../../lib/macro';
import { light, dark } from './constant'

const darkOrLight = (bool) => bool ? light : dark
const btn = css\`
.light {
  color: \${darkOrLight(true)};
}
.dark {
  color: \${darkOrLight(false)};
}
\`;

export default props => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_a from './.index.test.module.css'
import { light, dark } from './constant'

const darkOrLight = (bool) => (bool ? light : dark)

const btn = _cssed_a
export default (props) => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
)


`;

exports[`unknown plugin multi call macro: multi call macro 1`] = `

import { css } from '../../lib/macro';

const blue = css\`
.box {
  color: blue;
}
\`;

const red = css\`
.box {
  color: red;
}
\`;

export default props => (
  <div className={props.isRed ? red.box : blue.box} />
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_b from './.index.test.b.module.css'
import _cssed_a from './.index.test.a.module.css'
const blue = _cssed_a
const red = _cssed_b
export default (props) => <div className={props.isRed ? red.box : blue.box} />


`;

exports[`unknown plugin multi call with external dependency: multi call with external dependency 1`] = `

import { css } from '../../lib/macro';
import { light, dark } from './constant'

const btn = css\`
.light {
  color: \${light};
}
.dark {
  color: \${dark};
}
\`;

export default props => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_a from './.index.test.module.css'
import { light, dark } from './constant'
const btn = _cssed_a
export default (props) => (
  <>
    <div className={btn.light} />
    <div className={btn.dark} />
  </>
)


`;

exports[`unknown plugin single call macro: single call macro 1`] = `

import { css } from '../../lib/macro';

const styles = css\`
  .box {
    color: blue
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import _cssed_a from './.index.test.module.css'
const styles = _cssed_a


`;
